tensorboard/data/server/data_compat.rs

```diff
@@ -45,6 +47,52 @@ pub enum EventValue {
     Summary(SummaryValue),
 }
 
+impl EventValue {
+    /// Consumes this event value and enriches it into a scalar.
+    ///
+    /// This supports `simple_value` (TF 1.x) summaries as well as non-empty tensors of type
+    /// `DT_FLOAT` or `DT_DOUBLE`. If a tensor has more than one item, the first will be taken.
```

Comment:
I have also been wondering this for a while. I don’t know that it’s
right to say that we only _support_ f32—we can read f64s, right?—but
it’s certainly true that we only _write_ f32. So, given that you’ve also
raised this, I’m happy to be more restrictive here. Sending 20% fewer
bytes over the wire is always nice.

If you don’t mind, I’ll do this in a separate PR, since it’s slightly
cross-cutting.

> if we wanted to allow the possibility of writing custom summaries with
> the scalar plugin name and an f64 tensor value

Not just the `"scalars"` plugin name, right? Any generic scalar summary.

Reply:
Yeah, agreed on all counts! I have a commit drafted locally.